### DISCUSSION TITLE
Ability To Duplicate Stock Locations

### DISCUSSION THREAD

**Discussion by compumatter** [inventree/InvenTree#4220] [Q&A] [answered]
I am setting up locations for stock.  In my case I have about 20 shelving units, each identical with 5 shelves and horizontal dividers within those shelves.

Creating a stock location on one shelf is the equivalent of the following.  By the time I get done I've got a lot of data entry.  Is there a way to duplicate Shelving Unit A and its children?

Shelving Unit A
-- Shelf 1
--- Section 1 (2,3,4)
-- Shelf 2
--- Section 1
--- Section 2 (3,4)

Thanks.

Jay

**Comment by wolflu05** [comment, marked as answer]
Hello @compumatter ,

I haven't thought of deep-duplicating locations, but I'm currently working on a [plugin](https://github.com/wolflu05/inventree-bulk-plugin) which tries to simplify bulk creation of stock locations/part categories end even ensure the naming along your storage tree with saved templates. Maybe this would also help you? It's currently in a beta phase and I have a [breaking change](https://github.com/wolflu05/inventree-bulk-plugin/pull/7) planned. You can read more about my plugin in #4084 .

**Comment by compumatter** [comment]
Looks interesting.  I'll need a few days but I'll download and try it ou

**Comment by SchrodingersGat** [comment]
@compumatter if this initial setup is a "one time" thing, it might be simplest to script this using the [python interface](https://docs.inventree.org/en/latest/api/python/python/)?
